src/qt/radiobut.cpp

```diff
@@ -39,6 +40,37 @@ bool wxRadioButton::Create( wxWindow *parent,
              const wxString& name)
 {
     m_qtRadioButton = new QRadioButton( parent->GetHandle() );
+
+    if ( ( style & wxRB_GROUP ) || ( style & wxRB_SINGLE ) )
+    {
+        createAndJoinNewGroup();
+    }
+    else
+    {
+        wxWindowList::compatibility_iterator node = parent->GetChildren().GetLast();
```

Comment:
OTOH... this probably works fine, and a similar implementation is already present for MSW, but would you consider an alternative without dynamic casts?

i.e.
- add a different member to `wxRadioButton` class
`static std::map<wxWindow*, QButtonGroup*> m_lastGroup;`
- replace the implementation here with
```
if ( HasFlag(wxRB_GROUP) || HasFlag(wxRB_SINGLE) )
{
    // create its own group
    QButtonGroup* group = new QButtonGroup();
    group->addButton(m_qtRadioButton);

    if( HasFlag(wxRB_SINGLE) )
    {
        // future buttons can no longer be added to an existing group
        m_lastGroup.erase(parent);
    }
    else
    {
        // future buttons can be added to this group
        m_lastGroup[parent] = group;
        m_qtRadioButton->setChecked(true); // the first button in a group should be selected
    }
}

else
{
    // add it to the previous group, if any
    std::map<wxWindow*, QButtonGroup*>::iterator it = m_lastGroup.find(parent);
    if ( it != m_lastGroup.end() )
    {
        it->second->addButton(m_qtRadioButton);
    }
}
```
- add some cleanup to the destructor
```
if ( m_qtRadioButton->group() )
{
    // if this button is the only member of the last group, no longer keep track of the latter
    std::map<wxWindow*, QButtonGroup*>::iterator it = m_lastGroup.find(GetParent());
    if ( it != m_lastGroup.end() && it->second == m_qtRadioButton->group() &&
         m_qtRadioButton->group()->buttons().size() == 1 )
    {
        m_lastGroup.erase(it);
    }
}
```
What do you think?

p.s. 1. I didn't test this.
p.s. 2. Does QButtonGroup need the same parent as its buttons? Or Qt sets its parent when the first child is added?
p.s. 3. Other opinions on this are definitely welcome.

Reply:
To answer 2, I couldn't find anything definitive online but from my own testing it appears that QButtonGroup and the buttons it holds can have different parents.
 
The approach you suggested seems to work fine and I'm happy with it.